compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -795,7 +795,7 @@ trait Checking {
   private def checkLegalImportOrExportPath(path: Tree, kind: String)(using Context): Unit = {
     checkStable(path.tpe, path.srcPos, kind)
     if (!ctx.isAfterTyper) Checking.checkRealizable(path.tpe, path.srcPos)
-    if !isIdempotentExpr(path) then
+    if ctx.phase.isCheckable && !isIdempotentExpr(path) then
```

Comment:
Seems like something that should be a pre-condition even before getting into here - but that's cleanup that can follow-up.

Reply:
This is not necessary, it must have papered over another bug somewhere else early on in my testing. Thanks for pointing it out.